engine/class_modules/monk/sc_monk.cpp

```diff
@@ -8398,6 +8885,58 @@ void monk_t::init_special_effects()
     create_proc_callback( talent.master_of_harmony.aspect_of_harmony_heal, cb );
   }
 
+  // ======================================
+  // Courage of the White Tiger
+  // ======================================
+  if ( talent.conduit_of_the_celestials.courage_of_the_white_tiger->ok() )
+  {
+    create_proc_callback( talent.conduit_of_the_celestials.courage_of_the_white_tiger.spell(),
+                          []( monk_t *p, action_state_t *state ) {
+                            if ( (p->specialization() == MONK_MISTWEAVER && state->action->id == p->baseline.monk.vivify->id() ) ||
+                                state->action->id == p->baseline.monk.tiger_palm->id() )
+                              return true;
+
+                            return false;
+                          } );
+
+    callbacks.register_callback_execute_function(
+        talent.conduit_of_the_celestials.courage_of_the_white_tiger.spell()->id(),
+        [ this ]( const dbc_proc_callback_t *, action_t *, action_state_t *state ) {
+          active_actions.courage_of_the_white_tiger->set_target( state->target );
+        } );
+  }
+
+  // ======================================
+  // Flight of the Red Crane
+  // ======================================
+  if ( talent.conduit_of_the_celestials.flight_of_the_red_crane->ok() )
+  {
+    create_proc_callback(
+        talent.conduit_of_the_celestials.flight_of_the_red_crane.spell(), []( monk_t *p, action_state_t *state ) {
+          if ( ( p->specialization() == MONK_MISTWEAVER && state->action->id == p->talent.mistweaver.refreshing_jade_wind_tick->id() ) ||
+             ( p->specialization() == MONK_WINDWALKER && state->action->id == p->passives.rushing_jade_wind_tick->id() ) ||
+               state->action->id == p->baseline.monk.spinning_crane_kick->effectN( 1 ).trigger()->id() )
+            return true;
+
+          return false;
+        } );
+
+    callbacks.register_callback_execute_function(
+        talent.conduit_of_the_celestials.flight_of_the_red_crane.spell()->id(),
+        [ this ]( const dbc_proc_callback_t *, action_t *, action_state_t *state ) {
+          if (specialization() == MONK_MISTWEAVER)
+          {
+            active_actions.flight_of_the_red_crane_heal->set_target( state->target );
+            active_actions.flight_of_the_red_crane_heal->execute();
+          }
+          else
+          {
+            active_actions.flight_of_the_red_crane_damage->set_target( state->target );
+            active_actions.flight_of_the_red_crane_damage->execute();
+          }           
+        } );
+  }
```

Comment:
You could set the target in the lambda created in `create_proc_callback` args, no need to regsiter an execute function on top.

You can also opt to use `execute_on_target( player_t* )` for brevity.